frontend/src/BenchmarkData.res

```diff
@@ -0,0 +1,52 @@
+open Belt
+
+module PullNumberId = Id.MakeComparable({
+  type t = option<int>
+  let cmp = (a, b) => Pervasives.compare(a, b)
+})
+
+type timeseries = array<array<float>>
+type byMetricName = Map.String.t<(timeseries, array<{"commit": string, "runAt": Js.Date.t}>)>
+type byTestName = Map.String.t<byMetricName>
+type byPullNumber = Map.t<PullNumberId.t, byTestName, PullNumberId.identity>
+type t = byPullNumber
+
+let empty: t = Map.make(~id=module(PullNumberId))
+
+let add = (
+  byPullNumber: t,
+  ~pullNumber,
+  ~testName,
+  ~metricName,
+  ~runAt: Js.Date.t,
+  ~commit,
+  ~value: float,
+) => {
+  // Unwrap
+  let byTestName = Map.getWithDefault(byPullNumber, pullNumber, Map.String.empty)
+  let byMetricName = Map.String.getWithDefault(byTestName, testName, Map.String.empty)
+  let (timeseries, metadata) = Map.String.getWithDefault(byMetricName, metricName, ([], []))
+
+  // Update
+  let row = [(Obj.magic(runAt): float), value]
```

Comment:
We should probably add `LineGraph.Data` to manage this in a clean way.

For now it is safe because the underlying JS array has mixed types.